### PR TITLE
Don't emit profiler events we don't need to.

### PR DESCRIPTION
**Pull request by kumpera**
This PR those the following: 
- Avoid GC and thread events when not needed.
- Avoid GC events for ignored collections when using heapshot

This PR is based on https://github.com/mono/mono/pull/4967 so either ignore it for now or review only the top 2 commits. :)

### CODE REVIEW

**Comment by kumpera** [comment]
build pkg

File: mono/profiler/log.c [hunk 1]
```diff
@@ -1715,13 +1712,16 @@ finalize_begin (MonoProfiler *prof)
 static void
 finalize_end (MonoProfiler *prof)
 {
-	ENTER_LOG (&finalize_ends_ctr, buf,
-		EVENT_SIZE /* event */
-	);
+	trigger_on_demand_heapshot ();
+	if (ENABLED (MONO_PROFILE_GC_FINALIZATION)) {
```
Comment (alexrp) [inline mono/profiler/log.c]: Shouldn't this be `PROFLOG_FINALIZATION_EVENTS` ?

Comment (kumpera) [inline mono/profiler/log.c]: Yeah, I'll fix it.

File: mono/profiler/log.c [hunk 2]
```diff
@@ -1502,7 +1505,8 @@ gc_event (MonoProfiler *profiler, MonoGCEvent ev, int generation)
 		 * object allocation events for certain addresses could come
 		 * after the move events that made those addresses available.
 		 */
-		sync_point_mark (SYNC_POINT_WORLD_START);
+		if (ENABLED (PROFLOG_GC_EVENTS | PROFLOG_GC_MOVES_EVENTS))
```
Comment (alexrp) [inline mono/profiler/log.c]: Why are we checking both flags here?

**Pull request merged**